test/it/it/suites/DbBaseSpec.scala

```diff
@@ -0,0 +1,196 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package it.suites
+
+
+import cats.effect.{ContextShift, IO, Resource, Timer}
+import com.codahale.metrics.SharedMetricRegistries
+import com.dimafeng.testcontainers.{ForAllTestContainer, PostgreSQLContainer}
+import doobie._
+import doobie.implicits._
+import doobie.postgres.{PFCM, PHC}
+import doobie.util.fragment.Fragment.{const => csql}
+import org.scalatest.BeforeAndAfterAll
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.wordspec.AnyWordSpec
+import org.scalatestplus.play.guice.GuiceOneAppPerSuite
+import org.testcontainers.utility.DockerImageName
+import play.api
+import play.api.Application
+import play.api.inject.bind
+import play.api.inject.guice.GuiceApplicationBuilder
+
+import java.io.File
+import java.nio.file.{Files, StandardOpenOption}
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.io.Source
+
+trait DbBaseSpec extends AnyWordSpec with Matchers with BeforeAndAfterAll with ForAllTestContainer with GuiceOneAppPerSuite {
+  implicit val cs: ContextShift[IO] = IO.contextShift(global)
+  implicit val timer: Timer[IO] = IO.timer(global)
+
+  SharedMetricRegistries.clear()
+
+  protected val schemaName = "test_schema"
+
+  override lazy val container: PostgreSQLContainer = PostgreSQLContainer(
+    dockerImageNameOverride = DockerImageName.parse("postgres:10-alpine"),
+    databaseName = "addressbasepremium",
+    username = "root",
+    password = "Passw0rd123"
+  )
+
+  private def createTransactor(): IO[Transactor[IO]] = {
+    val ptx = Transactor.fromDriverManager[IO](
+      driver = "org.postgresql.Driver",
+      url = container.jdbcUrl,
+      user = container.username,
+      pass = container.password
+    )
+
+//    val ptx = Transactor.fromDriverManager[IO](
+//      driver = "org.postgresql.Driver",
+//      url = "jdbc:postgresql://localhost:5433/addressbasepremium",
+//      user = "root",
+//      pass = "password"
+//    )
+
+    IO(ptx)
+  }
+
+  private def createTestData(t: Transactor[IO]): IO[Transactor[IO]] = {
+    val sqlMap = loadSqlFiles
+
+    val createDbObjects: IO[Int] = (for {
+      _ <- csql(s"""CREATE SCHEMA ${schemaName}""").update.run
+      _ =  println(s""">>> Created schemaName: ${schemaName}""")
+      _ <- csql(sqlMap("createDbSchemaUrl")).update.run
+      _ =  println(s""">>> Created schema objects""")
+      _ <- csql("""CREATE TABLE IF NOT EXISTS public.address_lookup_status (
+                   | schema_name VARCHAR(64) NOT NULL PRIMARY KEY,
+                   | status      VARCHAR(32) NOT NULL,
+                   | error_message VARCHAR NULL,
+                   | timestamp   TIMESTAMP NOT NULL);""".stripMargin).update.run
+      _ =  println(s""">>> Created public.address_lookup_status table""")
```

Comment:
Have we got any check to ensure that this structure matches the real DB structure?

Reply:
The structure for the tables and views other than address_lookup_status is pulled directly from the lambda repo. These tests essentially check that the AddressSearchRepo works with the db schema as expected.

Reply:
There is another bit of work that could/should be done and that is to pull out all the sql into files in the ingest lambda repo. Then, this test will reference those directly and we won't need to inline the sql here + nothing to keep in sync.